### PR TITLE
try with mingw

### PR DESCRIPTION
**Pull request by stefanberger**
Signed-off-by: Stefan Berger <stefanb@linux.ibm.com>

### CODE REVIEW

**Comment by stefanberger** [comment]
@estesp @dims @mikebrow I am trying to figure out why containerd build is breaking. I added CGO_ENABLED=1 to the Windows build (in this patch) and it doesn't seem to break the build: https://github.com/containerd/imgcrypt/pull/39/checks?check_run_id=2022405348#step:5:67

Also see my comment here regarding how we fixed cross builds that did have problems with cgo: https://github.com/containerd/imgcrypt/issues/38

**Comment by mikebrow** [comment]
> @estesp @dims @mikebrow I am trying to figure out why containerd build is breaking. I added CGO_ENABLED=1 to the Windows build (in this patch) and it doesn't seem to break the build: https://github.com/containerd/imgcrypt/pull/39/checks?check_run_id=2022405348#step:5:67
> 
> Also see my comment here regarding how we fixed cross builds that did have problems with cgo: #38

@kevpar can you take a look at the runner here and contrast that to your go build environment that failed due to the cgo dependency?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -138,3 +138,32 @@ jobs:
           make test
           make
         working-directory: src/github.com/containerd/imgcrypt
+
+  windows-mingw:
+    name: Windows Tests
```
Comment (mikebrow) [inline .github/workflows/ci.yml:143]: nit... I'd call it Windows mingw32 Testing

will need Kevin to chime in on what if any windows tooling also needs to be obfuscated while go build is running to setup a reproduction to typical windows environments outside this runner

Comment (stefanberger) [reply .github/workflows/ci.yml:143]: Sure, I can rename this. For now we need to figure out what actually the culprit is that breaks the containerd build. I am not actually sure whether we gain any insights with this additional build...

**Comment by kevpar** [comment]
> > @estesp @dims @mikebrow I am trying to figure out why containerd build is breaking. I added CGO_ENABLED=1 to the Windows build (in this patch) and it doesn't seem to break the build: https://github.com/containerd/imgcrypt/pull/39/checks?check_run_id=2022405348#step:5:67
> > Also see my comment here regarding how we fixed cross builds that did have problems with cgo: #38
> 
> @kevpar can you take a look at the runner here and contrast that to your go build environment that failed due to the cgo dependency?

I'm pretty sure it works here because GitHub Actions windows-2019 includes the 64-bit MinGW version ([reference](https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md#tools)). When I tested locally I just had the 32-bit version installed (32-bit and 64-bit seem to be separate projects).

**Comment by stefanberger** [comment]
> I'm pretty sure it works here because GitHub Actions windows-2019 includes the 64-bit MinGW version ([reference](https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md#tools)). When I tested locally I just had the 32-bit version installed (32-bit and 64-bit seem to be separate projects).

Is 32bit an unsupported build environment or do we need to care? The question is whether we need to fix something in imgcrypt or ocicrypt related to your build failure, or we can move on from here?

**Comment by mikebrow** [comment]
Thx @stefanberger

**Comment by kevpar** [comment]
> > I'm pretty sure it works here because GitHub Actions windows-2019 includes the 64-bit MinGW version ([reference](https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md#tools)). When I tested locally I just had the 32-bit version installed (32-bit and 64-bit seem to be separate projects).
> 
> Is 32bit an unsupported build environment or do we need to care? The question is whether we need to fix something in imgcrypt or ocicrypt related to your build failure, or we can move on from here?

Sorry for not responding sooner, got distracted by some other things. I don't think it is a big deal to require mingw-w64 be installed to build containerd on Windows. The other option is to use WSL to build, so I don't think users are left without options here.

I did install mingw-w64 locally today and verified I can build containerd with it installed.

**Pull request closed without merging**